core/base/planarGraphLayout/MergeTreeVisu.h

```diff
@@ -0,0 +1,948 @@
+        shiftBranchBounds(retVec, treeSimplexId, branching, allBranchBounds,
+                          allBranchOrigins[branchNodeOrigin], tree,
+                          branchNodeOrigin, shift);
+      }
+
+      // Shift a branch if conflict with another one
+      for(unsigned int i = 1; i < allBranchOrigins[nodeOrigin].size(); ++i) {
+        ftm::idNode branchNodeOrigin = allBranchOrigins[nodeOrigin][i];
+        ftm::idNode branchNode = tree->getNode(branchNodeOrigin)->getOrigin();
+        for(unsigned int j = 0; j < i; ++j) {
+          auto first = allBranchBounds[branchNodeOrigin];
+          ftm::idNode previousBranchNodeOrigin
+            = allBranchOrigins[nodeOrigin][j];
+          auto second = allBranchBounds[previousBranchNodeOrigin];
+
+          bool branchConflict = isConflictingBranchAndBound(
+            first, second, tree, previousBranchNodeOrigin, retVec,
+            treeSimplexId);
+          if(isConflictingBounds(first, second) or branchConflict) {
+
+            // Get left or right orientation given the branch
+            int lastIndex = nodeBranching[branchNodeOrigin].size() - 1;
+            bool isLeft
+              = (retVec
+                   [treeSimplexId[nodeBranching[branchNodeOrigin][lastIndex]]
+                    * 2]
+                 //< retVec[treeSimplexId[nodeOrigin]*2]);
+                 < retVec[treeSimplexId[node] * 2]);
+
+            // Get shift
+            float branchNodeOriginXmax = std::get<1>(first);
+            float previousbranchNodeOriginXmin = std::get<0>(second);
+            // float branchNodeOriginXmin = std::get<0>(first);
+            float previousbranchNodeOriginXmax = std::get<1>(second);
+            float shift
+              = isLeft ? previousbranchNodeOriginXmin - branchNodeOriginXmax :
+                       // previousbranchNodeOriginXmax - branchNodeOriginXmin;
+                  previousbranchNodeOriginXmax
+                    - retVec[treeSimplexId[branchNode] * 2];
+            bool isSubBranchImportant = isImportantPair<dataType>(
+              tree, branchNodeOrigin, ImportantPairs);
+            shift += (isLeft ? -1 : 1)
+                     * (isSubBranchImportant ? importantPairsGap
+                                             : nonImportantPairsGap);
+            // shift += (isLeft ? -1 : 1) * nonImportantPairsGap;
+
+            // Shift bounds
+            shiftBranchBounds(retVec, treeSimplexId, branching, allBranchBounds,
+                              allBranchOrigins[branchNodeOrigin], tree,
+                              branchNodeOrigin, shift);
+          }
+        }
+      } // end for
+
+      // TODO optimize get branch bounds by using results previously computed
+      // Get branch x and y bounds
+      allBranchBounds[nodeOrigin]
+        = getBranchBounds(retVec, treeSimplexId, branching, tree, nodeOrigin);
+
+      // Verify conflict (testing)
+      /*printMsg( "====================================" ,
```

Comment:
Add comments to explain what this code snippet is doing and what is left to be done.

Reply:
Done!